File: tests/expected/rect.kt
// This file implements a rectangle class

data class Rectangle(
    val height: Int,
    val length: Int,
) {
    fun is_square(): Boolean = height == length
}

fun show() {
    var r = Rectangle(1, 1)
    assert(r.is_square())
    r = Rectangle(1, 2)
    assert(!(r.is_square()))
    if (true) {
        val __tmp1 = r.height
        println("$__tmp1")
    }
    if (true) {
        val __tmp2 = r.length
        println("$__tmp2")
    }
}

fun main(argv: Array<String>) {
    show()
}

File: tests/expected/rect.dart
// @dart=3.4
import 'package:sprintf/sprintf.dart';
/* This file implements a rectangle class  */

class Rectangle {
  int height;
  int length;

  Rectangle(this.height, this.length) {}
  bool is_square() {
    return height == length;
  }
}

show() {
  Rectangle r = Rectangle(1, 1);
  assert(r.is_square());
  r = Rectangle(1, 2);
  assert(!(r.is_square()));
  print(sprintf("%s", [r.height]));
  print(sprintf("%s", [r.length]));
}

main(List<String> argv) {
  show();
}
